cartographer_ros/cartographer_ros/assets_writer.cc

```diff
@@ -189,73 +191,100 @@ void AssetsWriter::Run(const std::string& configuration_directory,
       if (trajectory_proto.node_size() == 0) {
         continue;
       }
-      tf2_ros::Buffer tf_buffer;
+      std::shared_ptr<tf2_ros::Buffer> tf_buffer = std::make_shared<tf2_ros::Buffer>(clock);
```

Comment:
I don't really have a strong opinion about this, but any reason to change this to a `shared_ptr` rather than just use the existing stack object?

Reply:
The reason I changed it to shared_ptr is to refer to the case given on the [ROS2 documentation](https://docs.ros.org/en/foxy/Tutorials/Tf2/Writing-A-Tf2-Listener-Cpp.html#id4), but I also feel that there is no need to use shared_ptr here, and the use of stack object should run faster